x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/DeploymentManager.java

```diff
@@ -74,6 +77,7 @@ public class DeploymentManager {
 
     private static final Logger logger = LogManager.getLogger(DeploymentManager.class);
     private static final AtomicLong requestIdCounter = new AtomicLong(1);
+    public static final int NUM_START_ATTEMPTS = 3;
```

Comment:
Sorry, I didn't look closely enough first time. We try 4 times in total, so one start and three restarts.

```suggestion
    public static final int NUM_RESTART_ATTEMPTS = 3;
```